### PR TITLE
feat: allow blueprints to specify a custom resolution for NORA hover preview SOFIE-2840

### PR DESCRIPTION
**Pull request by Julusian**
## About the Contributor
This pull request is posted on behalf of the NRK.

## Type of Contribution

This is a:  Feature

## Current Behavior
The NORA hover preview is currently fixed at being run at a resolution of 1920x1080.

## New Behavior
The blueprints are able to specify a different resolution for the hover preview window, so that it can more accurately represent the content as it will be shown on video walls with other aspect ratios

## Testing Instructions
This requires custom blueprints which set the properties

## Status

- [x] PR is ready to be reviewed.
- [x] The functionality has been tested by the author.
- [x] Relevant unit tests has been added / updated.
- [ ] Relevant documentation (code comments, [system documentation](https://nrkno.github.io/sofie-core/)) has been added / updated.

### CODE REVIEW

File: packages/blueprints-integration/src/content.ts [hunk 1]
```diff
@@ -128,6 +128,7 @@ export interface NoraPayload {
 export interface NoraContent extends BaseContent {
 	payload: NoraPayload
 	previewRenderer: string
+	previewRendererDimensions?: { width: number; height: number }
```
Comment (nytamin) [inline packages/blueprints-integration/src/content.ts:131]:
```suggestion
	/** Dimensions of the rendered hover-preview viewport in pixels. Defaults to 1920x1080. */
	previewRendererDimensions?: { width: number; height: number }
```

File: meteor/client/ui/FloatingInspectors/NoraFloatingInspector.tsx [hunk 2]
```diff
@@ -152,32 +152,50 @@ export class NoraPreviewRenderer extends React.Component<{}, IStateHeader> {
 		this.rootElement = e
 	}
 
-	private getElStyle() {
+	private getElStyle(width: number, height: number) {
 		const style = { ...this.state.style }
 		style.visibility = this.state.show ? 'visible' : 'hidden'
+		style.width = `${width / 4}px`
+		style.height = `${height / 4}px`
```
Comment (nytamin) [inline meteor/client/ui/FloatingInspectors/NoraFloatingInspector.tsx:159]: I think we should make the viewed size be independent of the absolute rendered values.
So something like this instead?

```typescript
const ratio = height / width
const idealWidth = Math.max(300, browserWidth/4)
style.width = `${idealWidth}px`
style.height = `${idealWidth * ratio}px`

```

**Review by nytamin — CHANGES_REQUESTED** [review]

File: meteor/client/ui/FloatingInspectors/NoraFloatingInspector.tsx (hunk 2 above)
Comment (Julusian) [reply meteor/client/ui/FloatingInspectors/NoraFloatingInspector.tsx:159]: I am open to this, but I think it needs a team/UX discussion to figure out what rule it should follow.

Comment (Julusian) [reply meteor/client/ui/FloatingInspectors/NoraFloatingInspector.tsx]: After discussion with the team, we agreed on making the preview be a maximum of 480px in either dimension

**Pull request merged**